flow/serialize.h

```diff
@@ -665,8 +665,12 @@ class BinaryReader {
 	ProtocolVersion m_protocolVersion;
 };
 
-struct SendBuffer {
-	uint8_t const* data;
+class SendBuffer {
+	uint8_t* _data;
+
+public:
+	inline uint8_t const* data() const { return _data; }
+	inline uint8_t*& data() { return _data; }
```

Comment:
This is scary, since now someone can typo and change our buffer to some buffer of unknown lifetime.  What if data were protected, and this returned `uint8_t*` instead?